### PR TITLE
Add means to shim out sbt-project-matrix for IntelliJ import

### PR DESCRIPTION
**Pull request by retronym**

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=pull/890/head&io=develop&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

**Comment by retronym** [comment]
The other problem I have with Zinc + IntelliJ is that zinc-core-persist is not imported as a dependency of zinc-core, so e.g. `ProtobufReaders.scala` can't resolve `Scema.UsedName` etc.

File: project/ProjectMatrixShims.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import sbt.{Keys, LocalProject, Project, ProjectRef}
+import scala.language.experimental.macros
+
+
+object ProjectMatrixShims {
+
+  implicit def RichProject(p: Project) = new RichProject(p)
+
+  class RichProject(p: Project) {
+    def configure(fs: List[(Project) => Project]): Project = fs.foldRight(p)(_.apply(_))
```
Comment (dwijnand) [inline project/ProjectMatrixShims.scala]: This one's native to Project, IIRC. (nit)

**Review by dwijnand — APPROVED** [review]

**Comment by dwijnand** [comment]
> The other problem I have with Zinc + IntelliJ is that zinc-core-persist is not imported as a dependency of zinc-core, so e.g. `ProtobufReaders.scala` can't resolve `Scema.UsedName` etc.

Yeah, same very familiar.

**Comment by dwijnand** [comment]
In the same vein: #892

**Review by eed3si9n — APPROVED** [review]
Fine by me

**Comment by SethTisue** [comment]
hopefully #1075 is a better approach going forward

**Pull request merged**